src/IO.cpp

```diff
@@ -224,13 +224,13 @@ ObservationsForInference ReadInferenceObservations(const std::string& configDir,
     int nHealthBoards = validationParams.nHealthBoards;
     int nCasesDays = validationParams.nCasesDays;
 
-    unsigned int cases_rows = observations.cases.size();
+    unsigned int cases_rows = Utilities::checkAndGetSize(observations.cases, "observations.cases");
```

Comment:
I'm a little confused by this. It seems that `checkAndGetSize` is just doing a subset of what the `ImportConsistencyCheck` function is doing a few lines down? (The former only checks that there are a non-zero number of rows in `observations.cases`; the latter checks that there are exactly the correct number of rows and columns).

Reply:
The current ordering will cause a memory issue if the number of rows is zero:

```
unsigned int cases_rows = observations.cases.size();
unsigned int cases_cols = observations.cases[0].size();
```
I agree my method might not be needed but if you plan to drop it then we need to reorder so that the rows are checked before the columns are retrieved, i.e.:
```
unsigned int cases_rows = observations.cases.size();
ImportConsistencyCheck(scot_data_file, cases_rows, (nHealthBoards + 1), "rows");
unsigned int cases_cols = observations.cases[0].size();
ImportConsistencyCheck(scot_deaths_file, deaths_cols, nCasesDays, "columns");
```
instead.